### PR TITLE
Fix the empty price_s case and self.instruments in SBBStrategyEMA.

### PR DESCRIPTION
**Pull request by ChiahungTai**
## Description

1. Deal with price_s.empty:
2. Deal with instruments is List

## How Has This Been Tested?

- [x] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [ ] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
3. Your own tests:

## Types of changes

- [x] Fix bugs
- [ ] Add new feature
- [ ] Update documentation

### CODE REVIEW

**Comment by ChiahungTai** [comment]
@Fivele-Li @you-n-g,
Looks like the pylint has some problems, I think that is not my code's problem. Can you guys give me some suggestions to fix it?

File: qlib/contrib/strategy/rule_strategy.py [hunk 1]
```diff
@@ -328,6 +328,8 @@ def __init__(
             self.instruments = "all"
         if isinstance(instruments, str):
             self.instruments = D.instruments(instruments)
+        if isinstance(instruments, List):
```
Comment (you-n-g) [inline qlib/contrib/strategy/rule_strategy.py:331]:
```suggestion
        elif isinstance(instruments, List):
```

File: qlib/contrib/strategy/rule_strategy.py [hunk 2]
```diff
@@ -328,6 +328,8 @@ def __init__(
             self.instruments = "all"
         if isinstance(instruments, str):
```
Comment (you-n-g) [inline qlib/contrib/strategy/rule_strategy.py:329]:
```suggestion
        elif isinstance(instruments, str):
```

**Comment by SunsetWolf** [comment]
Thank you for your contribution. Now merged.

**Pull request merged**